src/game_party.cpp

```diff
@@ -707,3 +700,9 @@ Game_Actor* Game_Party::GetHighestLeveledActorWhoCanUse(const RPG::Item* item) c
 	}
 	return best;
 }
+
+std::pair<int,bool> Game_Party::GetItemIndex(int item_id) const {
+	auto& ids = data().item_ids;
+	auto iter = std::lower_bound(ids.begin(), ids.end(), item_id);
+	return std::make_pair(iter - ids.begin(), (iter != ids.end() && *iter == item_id));
```

Comment:
Why do you use std::lower_bound and not std::find here when the goal is to find the item with the specific id.

Reply:
lower_bound does a binary search, which find the item in lg(N) time instead of linear time for find.

Reply:
and std::binary_search returns bool instead of an iterator, hmpf.

Reply:
Oh yeah, that one is kind of useless.. Unless all you want to do is ask "Does this thing exist in the set"? Which is rare.